client/src/store/contentSlice.js

```diff
@@ -385,42 +418,107 @@ export const contentSlice = createSlice({
         state.countryOriginContentLoading = false;
       })
 
-      //fetch watch content
+      // Store watch history content
       .addCase(fetchContentByWatchHistory.pending, (state) => {
-        state.watchContentLoading = true;
+        state.watchHistoryLoading = true;
       })
       .addCase(fetchContentByWatchHistory.fulfilled, (state, action) => {
-        state.watchedContent = action.payload;
-        state.watchContentLoading = false;
+        state.watchHistoryContent = action.payload;
+        state.watchHistoryLoading = false;
       })
       .addCase(fetchContentByWatchHistory.rejected, (state) => {
-        state.watchedContent = [];
-        state.watchContentLoading = false;
+        state.watchHistoryContent = [];
+        state.watchHistoryLoading = false;
       })
 
-      //add watch content
+      // Add watch history content
       .addCase(addContentToWatchHistory.pending, (state) => {
-        state.watchContentLoading = true;
+        state.watchHistoryLoading = true;
       })
       .addCase(addContentToWatchHistory.fulfilled, (state, action) => {
         const watchContentId = action.payload;
 
-        const watchContent = state.watchedContent.find(
+        const isExitWatchContent = state.watchHistoryContent.find(
           (item) => item.contentId === watchContentId
         );
 
-        if (!watchContent) {
-          state.watchedContent.push(watchContent);
+        if (!isExitWatchContent) {
+          const watchContent = state.filteredContent.find(
+            (item) => item.contentId === watchContentId
+          );
+
+          if (!watchContent) {
+            state.watchHistoryContent.push(watchContent);
+          }
         }
 
-        state.watchedContent = state.watchContentLoading = false;
+        state.watchHistoryLoading = false;
       })
       .addCase(addContentToWatchHistory.rejected, (state) => {
-        state.watchedContent = [];
-        state.watchContentLoading = false;
+        state.watchHistoryContent = [];
+        state.watchHistoryLoading = false;
       })
 
-      //like content
+      // Store watch list content
+      .addCase(fetchContentByWatchList.pending, (state) => {
+        state.watchListLoading = true;
+      })
+      .addCase(fetchContentByWatchList.fulfilled, (state, action) => {
```

Comment:
I think the naming can be improved, maybe something like getWatchlistData or getWatchlistContent?

Reply:
Talk to my lead regarding this, because we all have to follow for same.